java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java

```diff
@@ -1774,7 +1774,7 @@ private long readVarUint36Slow() {
   /** Reads the 1-5 byte int part of a non-negative varint. */
   public int readVarUint32() {
     int readIdx = readerIndex;
-    if (size - readIdx < 5) {
+    if (size - readIdx < 4) {
```

Comment:
later `result |= (UNSAFE.getByte(heapMemory, address + readIdx++) & 0x7F) << 28;` wqill read one more byte without check. If we lower threshold, we may get crash in such cases.

Reply:
Could you add a test to cover such cases?

Reply:
> later result |= (UNSAFE.getByte(heapMemory, address + readIdx++) & 0x7F) << 28; wqill read one more byte without check. If we lower threshold, we may get crash in such cases.

The premise for executing `result |= (UNSAFE.getByte(heapMemory, address + readIdx++) & 0x7F) << 28` is that the fourth byte of `fourByteValue` is `0b1xxxxxxxx` (that is, there are more bytes to be parsed).

In `readVarUint36Slow()`, it is also possible to read 5 bytes continuously (provided that `(b & 0x80) != 0`).

Therefore, the premise of executing `result |= (UNSAFE.getByte(heapMemory, address + readIdx++) & 0x7F) << 28` is that we still have content that can be read in the Buffer. If there is no content to read, it means that an error occurred during writing.

Reply:
As you can see, we have `address + readIdx++`. It will read the 5rd bytes without range check. This will crash jvm if we don't have such big buffer or data is maliciously constructed

Reply:
So the five consecutive `getByte` calls in `readVarUint36Slow` are no different from `UNSAFE.getInt + UNSAFE.getByte`. They all have read checks for `(fourByteValue & 0x80000000) != 0` and `(b & 0x80) != 0`.

We only need to ensure that `size - readIdx >= 4` to use `UNSAFE.getInt` to read the content normally.

Reply:
You read five bytes, but you just check 4 bytes here

Reply:
The readVarUint36Slow is very rare, only last five bytes of the whole binary will go to this path. So we try to reduce the check in critical path

Reply:
As discussed in WeChat, this solution only benefits the last 4 bytes, but it will introduce extra code, causing the bytecode of this method to become larger. In most cases, this optimization will not take effect, so close this PR.